src/app/actions/upload.ts

```diff
@@ -0,0 +1,37 @@
+import httpStatusCode from 'http-status-codes';
+
+import { parseCSVFormData } from '@/lib/file';
+import { createSearchResult } from '@/repositories/searchResult';
+
+export async function handleCsv(file: FormDataEntryValue | null, userId: number) {
+  try {
+    if (!file) {
+      const error = { code: httpStatusCode.BAD_REQUEST, message: 'File must be sent' };
+
+      throw error;
+    }
+
+    const tags = await parseCSVFormData(file);
```

Comment:
I am unsure why you kept wanting to name this `tags`, wouldn't the following make more sense 😅 

```suggestion
    const keywords = await parseCSVFormData(file);
```

Reply:
Fixed in https://github.com/nimblehq/ic-nextjs-prabesh-dahal/pull/59/commits/60c2b4b73860a885a6a8b0065d51e0ec37930de6